### PR TITLE
Update SPIRE101 Walkthrough

### LINKED ISSUE #1815: Update SPIRE101 walkthrough
The [SPIRE101](https://github.com/spiffe/spire/blob/master/doc/SPIRE101.md) walkthrough is based on a docker-compose file that does not exist anymore. Update it to use the current targets available in the `Makefile`.

### PR DESCRIPTION
**Pull request by amartinezfayo**
Update SPIRE101 Walkthrough

Fixes #1815

### CODE REVIEW

File: doc/SPIRE101.md [hunk 1]
```diff
@@ -38,35 +38,27 @@ Note: If you don't already have Docker installed, please follow these [installat
...
 ## Walkthrough
 
-1.  Start the SPIRE container using docker-compose.
+1.  Build the development Docker image.
 
-        docker-compose up -d
+        make dev-image
 
-2.  Check to make sure the spire-dev docker container is running.
+2.  Run a shell in the development Docker container.
```
Comment (azdagron) [inline doc/SPIRE101.md:45]: Are these first two steps even necessary?  `make build` will download the necessary toolchain required to build the binaries...

Comment (amartinezfayo) [reply doc/SPIRE101.md:45]: I wanted to keep this walkthrough using a dev container, so you can add a user to play with workload attestation without altering your environment. What do you think?

Comment (azdagron) [reply doc/SPIRE101.md:45]: sgtm!

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**